src/state/slices/opportunitiesSlice/opportunitiesApiSlice.ts

```diff
@@ -151,131 +150,133 @@ export const opportunitiesApi = createApi({
         return { data: null }
       },
     }),
-    getOpportunityUserData: build.query<GetOpportunityUserDataOutput, GetOpportunityUserDataInput>({
-      queryFn: async (
-        { accountId, opportunityId, defiType, defiProvider },
-        { dispatch, getState },
-      ) => {
-        try {
-          const resolver = getUserDataResolversByDefiProviderAndDefiType(defiProvider, defiType)
-
-          if (!resolver) {
-            throw new Error(`resolver for ${DefiProvider.UniV2}::${defiType} not implemented`)
-          }
-
-          const { chainId: accountChainId } = fromAccountId(accountId)
-          const opportunityChainId = opportunityIdToChainId(opportunityId)
-          if (opportunityChainId !== accountChainId) {
-            const byAccountId: OpportunityDataById = {
-              [accountId]: [],
-            }
+    getOpportunityUserData: build.query<null, GetOpportunityUserDataInput[]>({
```

Comment:
This one is only fired for a single opportunity (ETH/Fox Staking) though I believe we might as well go with this refactor for consistency 🤌